src/Groups/matrices/matrix_manipulation.jl

```diff
@@ -238,6 +238,42 @@ function ishermitian_matrix(B::MatElem{T}) where T <: FinFieldElem
    return true
 end
 
+# return (true, h) if y = hx, (false, nothing) otherwise
+# FIXME: at the moment, works only for fields
+function _is_scalar_multiple_mat(x::MatElem{T}, y::MatElem{T}) where T <: RingElem
+   F=base_ring(x)
+   F==base_ring(y) || return (false, nothing)
+   nrows(x)==nrows(y) || return (false, nothing)
+   ncols(x)==ncols(y) || return (false, nothing)
+
+   if x==0
+      if y==0 return (true, F(1))
+      else return (false, nothing)
+      end
+   end
+
+
+   found = false
+   for i in 1:nrows(x)
+   for j in 1:ncols(x)
+      if x[i,j] !=0
+         global h = y[i,j] * x[i,j]^-1
+         found = true
+         break
+      end
+   end
+   if found==true break end
+   end
```

Comment:
```suggestion
   h = F(1)
   for i in 1:nrows(x), j in 1:ncols(x)
      if !iszero(x[i,j])
         iszero(y[i,j]) && return (false, nothing)
         h = y[i,j] / x[i,j]
         break
      end
   end
```

So if `x` is a zero matrix, this ends up with `h == F(1)`, and then the next loop compares the two matrices accordingly.

Reply:
Overall, though, I wonder if there is already a function in AA/Nemo/Hecke to find the first non-zero entry in a row or even a matrix (as one needs this quite often in various applications), which then might even be faster for some types of matrices... @fieker ?

Reply:
@thofma Told me that there is not such a function, so I wrote it. But the same argument holds for every function defined in this file: when similar methods are defined in AA/Hecke/Nemo, this one can be replaced.